index.js

```diff
@@ -12,7 +12,26 @@ function isNegative(pattern) {
 	return pattern[0] === '!';
 }
 
+function isString(value) {
+	return typeof value === 'string';
+}
+
+function assertPatternsInput(patterns) {
+	if (isString(patterns)) {
+		// bail early, the rest will easier this way
+		return;
+	}
+
+	if (!Array.isArray(patterns) || !patterns.every(isString)) {
+		throw new TypeError('patterns must be a string or an array of strings');
+	}
+}
+
 function generateGlobTasks(patterns, opts) {
+	// make sure input is correct, so that we can safely parse
+	// it in this task
+	assertPatternsInput(patterns);
+
 	var globTasks = [];
 
 	patterns = arrify(patterns);
```

Comment:
@catdad Please see the code sample that @schnittstabil posted above. I believe he suggested to remove the `arrify` dependency and manually array-ify the the patterns:

```js
patterns = Array.isArray(patterns) ? patterns : [patterns];
```

Before passing them to `assertPatternsInput`. Then you can remove the `isString(patterns)` and `!Array.isArray(patterns)` checks, just a `patterns.every(isString)` inside `assertPatternsInput` will suffice.